goqtdrv5/cdrv.h

```diff
@@ -217,7 +222,69 @@ inline QByteArray drvGetByteArray(void *param)
         return QByteArray();
     }
     slice_head *sh = (slice_head*)param;
-    return QByteArray(sh->data,sh->len);
+    return QByteArray(sh->data, sh->len);
+}
+
+inline void drvSetByteArray(void *param, const QByteArray& array)
+{
+    if (param == 0) {
+        return;
+    }
+
+    slice_head *sh = (slice_head*)param;
+    sh->len = array.size();
+    sh->data = array.data();
+}
+
+inline QStringList drvGetStringArray(void *param)
```

Comment:
Apparently, the definition of string_head is wrong:
```c++
struct string_head {
    char *data;
    goInt len;
    goInt cap; // This is due to being mapped to utf8_info, so we need to make sure we've got enough space
};
```
There shouldn't be a cap field in [StringHeader](http://golang.org/pkg/reflect/#StringHeader).
Let's remove this extra field. It's related to a hack in utf8_info defined in cdrv.go which uses a []byte to represent a string; we can redefine utf8_info to be a string and modify utf8_info_copy accordingly.